scripts/src/javatests/org/oppia/android/scripts/todo/CommentCheckTest.kt

```diff
@@ -0,0 +1,94 @@
+
+/** Tests for [CommentCheck]. */
+class CommentCheckTest {
+  private val permalinkPrefix = "https://github.com/oppia/oppia-android/blob/"
+  private val dummySha1 = "51ab6a0341cfb86d95a387438fc993b5eb977b83"
+  private val dummySha2 = "74cd6a0341cfb86d95a387438fc993b5eb977b83"
+
+  @Rule
+  @JvmField
+  var tempFolder = TemporaryFolder()
+
+  /**
+   * This is the case when no comment is present on the issue. In this case the Github API results
+   * in an empty comment body.
+   */
+  @Test
+  fun testCommentCheck_emptyLatestCommentBody_commentCheckShouldPass() {
+    val latestCommentFile = tempFolder.newFile("latest_comment.txt")
+    val scriptFailureCommentFile = tempFolder.newFile("todo_list.txt")
+    val latestCommentContent = ""
+    val scriptFailureCommentContent =
+      """
+      The issue is reopened because of the following unresolved TODOs:
+      $permalinkPrefix$dummySha1/scripts/BUILD.bazel#L184
+      $permalinkPrefix$dummySha1/scripts/BUILD.bazel#L185
+      
+      """.trimIndent()
+    latestCommentFile.writeText(latestCommentContent)
+    scriptFailureCommentFile.writeText(scriptFailureCommentContent)
+
+    main(tempFolder.root.toString(), "latest_comment.txt", "todo_list.txt")
+  }
+
+  @Test
+  fun testCommentCheck_differentComments_commentCheckShouldPass() {
+    val latestCommentFile = tempFolder.newFile("latest_comment.txt")
+    val scriptFailureCommentFile = tempFolder.newFile("todo_list.txt")
+    val latestCommentContent =
+      """
+      The issue is reopened because of the following unresolved TODOs:
+      $permalinkPrefix$dummySha1/scripts/BUILD.bazel#L184
+      $permalinkPrefix$dummySha1/scripts/BUILD.bazel#L185
+      $permalinkPrefix$dummySha1/scripts/BUILD.bazel#L186
+      
+      
+      """.trimIndent()
+    val scriptFailureCommentContent =
+      """
+      The issue is reopened because of the following unresolved TODOs:
+      $permalinkPrefix$dummySha2/scripts/BUILD.bazel#L184
+      $permalinkPrefix$dummySha2/scripts/BUILD.bazel#L185
+      
+      """.trimIndent()
+    latestCommentFile.writeText(latestCommentContent)
+    scriptFailureCommentFile.writeText(scriptFailureCommentContent)
+
+    main(tempFolder.root.toString(), "latest_comment.txt", "todo_list.txt")
+  }
+
+  @Test
+  fun testCommentCheck_sameComment_commentCheckShouldFail() {
+    val latestCommentFile = tempFolder.newFile("latest_comment.txt")
+    val scriptFailureCommentFile = tempFolder.newFile("todo_list.txt")
+    val latestCommentContent =
+      """
+      The issue is reopened because of the following unresolved TODOs:
+      $permalinkPrefix$dummySha1/scripts/BUILD.bazel#L184
+      $permalinkPrefix$dummySha1/scripts/BUILD.bazel#L185
+      
+      
+      """.trimIndent()
+    val scriptFailureCommentContent =
+      """
+      The issue is reopened because of the following unresolved TODOs:
+      $permalinkPrefix$dummySha2/scripts/BUILD.bazel#L184
+      $permalinkPrefix$dummySha2/scripts/BUILD.bazel#L185
```

Comment:
Related to an earlier comment, but I suggest having a distinct test for the sha logic vs. hiding it as part of this test.

Reply:
Done.